.github/pre-commit

```diff
@@ -8,6 +8,18 @@
 
 echo Shell: $SHELL
 
+## this will retrieve all of the .yaml files that have been
+## changed since the last commit
+STAGED_YAML_FILES=$(git diff --cached --name-only -- '*.yaml')
+
+## we can check to see if this is empty
+if [[ $STAGED_YAML_FILES == "" ]]; then
+    echo "No Yaml Files to Update"
+## otherwise we generate the manifests for master
+else
+    make manifests
+fi
```

Comment:
Logging that no changes happended isn't necessary imo. Consider changing it to something like this.

```suggestion
if [[ $STAGED_YAML_FILES != "" ]]; then
    echo "YAML files have changed, generating manifests"
    make manifests
fi
```